### PR TITLE
279 add new method getroibyid on roimapmanager

### PR DESCRIPTION
**Pull request by EscapedGibbon**

### CODE REVIEW

**Comment by targos** [comment]
In general, please take the time to write a pull request description in English (do not use the default one based on the branch name)

File: src/roi/RoiMapManager.ts [hunk 1]
```diff
@@ -66,6 +66,15 @@ export class RoiMapManager implements RoiManager {
     return this.map.data[this.map.width * row + column];
   }
 
+  public getRoiById(roiID: number): Roi {
+    const rois = this.getRois();
+    const thatRoi = rois.find((roi) => roi.id === roiID);
```
Comment (lpatiny) [inline src/roi/RoiMapManager.ts:71]: Could be renamed to `foundRoi`

File: src/roi/RoiMapManager.ts [hunk 2]
```diff
@@ -66,6 +66,15 @@ export class RoiMapManager implements RoiManager {
...
+    if (thatRoi === undefined) {
+      throw new Error(`invalid ID: ${roiID}`);
+    } else {
```
Comment (lpatiny) [inline src/roi/RoiMapManager.ts:74]: The else is not necessary

File: src/roi/RoiMapManager.ts [hunk 3]
```diff
@@ -66,6 +66,15 @@ export class RoiMapManager implements RoiManager {
     return this.map.data[this.map.width * row + column];
   }
 
+  public getRoiById(roiID: number): Roi {
+    const rois = this.getRois();
+    const thatRoi = rois.find((roi) => roi.id === roiID);
+    if (thatRoi === undefined) {
```
Comment (lpatiny) [inline src/roi/RoiMapManager.ts:72]: In this case I would write
```
if(! thatRoi) {

}
```

**Review by lpatiny — CHANGES_REQUESTED** [review]

**Pull request closed without merging**